### PR TITLE
Switch to sass-embedded and update eslint and stylelint

### PR DESCRIPTION
**Pull request by NeOMakinG**
| Questions         | Answers
| ------------------| -------------------------------------------------------
| Description?      | We need to update from node-sass to sass-embedded to fix some compatibilities issues and improve build performances, at the same time, I updated eslint dependencies
| Type?             | improvement
| BC breaks?        | no
| Deprecations?     | no
| How to test?      | Should build as usual with node 14 and 16 and styles should be fine on FO
| Possible impacts? | Dev tool and styling

### CODE REVIEW

**Comment by matks** [comment]
@NeOMakinG Should we mark this as a BC break for next release, and mention in a Changelog (like "BE CAREFUL! We changed THIS STUFF") ?

> We need to update from node-sass to sass-embedded to fix some compatibilities issues and improve build performances, at the same time, I updated eslint dependencies

**Comment by NeOMakinG** [comment]
> @NeOMakinG Should we mark this as a BC break for next release, and mention in a Changelog (like "BE CAREFUL! We changed THIS STUFF") ?
> 
> > We need to update from node-sass to sass-embedded to fix some compatibilities issues and improve build performances, at the same time, I updated eslint dependencies

This change shouldn't impact anything on user side, it shouldn't break the front-end part, but it might require you to run a newer node version to build the theme, I guess it shouldn't be a BC at all

**Review by atomiix — APPROVED** [review]

**Review by HanaRebaiQA — COMMENTED** [review]
Hello @NeOMakinG 

I have checked your PR. Here is my feedback.

- Check with `nodejs v16` and `npm v8` : An error is generated when run the **npm install && npm run build** under `classic/ _dev/ ` folder

image

When checking the FO : NOK

image

- Check with `nodejs v14` and `npm v6` : An error related to npm is generated when run the **npm install && npm run build** under `classic/ _dev/ ` folder

image

Could you please check ?

Thank You!

**Comment by kpodemski** [comment]
Hi @NeOMakinG 

Do you think it's doable to do that for the next version of Classic?

**Comment by NeOMakinG** [comment]
> Hi @NeOMakinG
> 
> Do you think it's doable to do that for the next version of Classic?

why not?

**Comment by leemyongpakvn** [comment]
@NeOMakinG @HanaRebaiQA For checking with `nodejs v16` and `npm v8`, it looks like a problem with missing sass-embedded-linux-x64, you can fix it by `npm i sass-embedded-linux-x64`.
I see no problem with a fresh install of PS 8.0.1, classic-theme 2.0.6 base (manual copy PR changes :) in  `nodejs v16` and `npm v8` environment.

**Comment by NeOMakinG** [comment]
@leemyongpakvn I suggest not doing this `npm i sass-embedded-linux-x64` it should work without it, I suspect some cache issues

**Comment by leemyongpakvn** [comment]
@NeOMakinG It is just a quick fix. We can remove the whole node_modules directory then `npm install` instead.

File: .github/workflows/lint.yml [hunk 1]
```diff
@@ -10,7 +10,7 @@ jobs:
       - name: Setup Node
         uses: actions/setup-node@v1
```
Comment (leemyongpakvn) [inline .github/workflows/lint.yml:11]: actions/setup-node@v3

File: .github/workflows/lint.yml [hunk 2]
```diff
@@ -23,7 +23,7 @@ jobs:
...
```
Comment (leemyongpakvn) [inline .github/workflows/lint.yml:24]: actions/setup-node@v3

**Review by leemyongpakvn — COMMENTED** [review]
It's right time to update actions/setup-node to v3.

**Pull request closed without merging**